pkg/controller/memberstatus/memberstatus_controller.go

```diff
@@ -200,11 +205,101 @@ func (r *ReconcileMemberStatus) memberOperatorHandleStatus(reqLogger logr.Logger
 	return err
 }
 
+// loadCurrentResourceUsage loads the current usage of the cluster and stores it into the member status
+func (r *ReconcileMemberStatus) loadCurrentResourceUsage(reqLogger logr.Logger, memberStatus *toolchainv1alpha1.MemberStatus) error {
+	memberStatus.Status.ResourceUsage.MemoryUsagePerNodeRole = map[string]int{}
+	allocatableValues, err := r.getAllocatableValues(reqLogger)
+	if err != nil {
+		return err
+	}
+
+	nodeMetricsList := &v1beta1.NodeMetricsList{}
+	if err := r.client.List(context.TODO(), nodeMetricsList); err != nil {
+		return err
+	}
+
+	usagePerRole := map[string]float32{}
+	allocatablePerRole := map[string]float32{}
+NodeMetricsList:
+	for _, nodeMetric := range nodeMetricsList.Items {
+		for key, usage := range nodeMetric.Usage {
+			if key == "memory" {
+				nodeInfo, found := allocatableValues[nodeMetric.Name]
+				if found {
+					usagePerRole[nodeInfo.role] += float32(usage.Value())
+					allocatablePerRole[nodeInfo.role] += float32(nodeInfo.allocatable.Value())
+
+					delete(allocatableValues, nodeMetric.Name)
+				} else {
+					reqLogger.Info("skipping NodeMetrics resource - there wasn't found corresponding node that would be monitored", "name", nodeMetric.Name)
+				}
+				continue NodeMetricsList
+			}
+		}
+		return fmt.Errorf("memory item not found in NodeMetrics: %v", nodeMetric)
+	}
+
+	if len(allocatableValues) > 0 {
+		var nodeNames []string
+		for nodeName := range allocatableValues {
+			nodeNames = append(nodeNames, nodeName)
+		}
+		return fmt.Errorf("missing NodeMetrics resource for Nodes: %v", nodeNames)
+	}
+
+	for role, usage := range usagePerRole {
+		memberStatus.Status.ResourceUsage.MemoryUsagePerNodeRole[role] = int((usage / allocatablePerRole[role]) * 100)
+	}
+	return nil
+}
+
+func (r *ReconcileMemberStatus) getAllocatableValues(reqLogger logr.Logger) (map[string]nodeInfo, error) {
+	nodes := &corev1.NodeList{}
+	err := r.client.List(context.TODO(), nodes)
+	if err != nil {
+		return nil, errs.Wrapf(err, "unable to list Nodes")
+	}
+	allocatableValues := map[string]nodeInfo{}
+	for _, node := range nodes.Items {
+		role := getNodeRole(node)
+		if role == "" {
+			reqLogger.Info("The node doesn't have role worker nor master - is ignored in resource consumption computing", "nodeName", node.Name)
+			continue
+		}
+		for key, value := range node.Status.Allocatable {
```

Comment:
same here, I assume you could use `if memory, exists := node.Status.Allocatable["memory"]; exists { ... }`

Reply:
done in 53280b6